ImageLounge/src/DkGui/DkThumbsWidgets.cpp

```diff
@@ -1224,19 +1226,39 @@ void DkThumbScene::keyPressEvent(QKeyEvent *event)
 
 void DkThumbScene::showFile(const QString &filePath)
 {
-    if (filePath == QDir::currentPath() || filePath.isEmpty()) {
-        int sf = getSelectedFiles().size();
-
-        QString info;
+    int sf = getSelectedFiles().size();
+    DkStatusBar *bar = DkStatusBarManager::instance().statusbar(); 
+    if (filePath == QDir::currentPath() || filePath.isEmpty()) { // i.e. user is NO LONGER hovering over a file
+        if (sf == 0) {
+            QString info = QString::number(mThumbLabels.size()) + tr(" images");
+            bar->setMessage(tr("%1 | %2").arg(info, currentDir()));
+            bar->setMessage("", DkStatusBar::status_filesize_info);
+        } else if (sf == 1) {
+            displayFileInfoInStatusbar(getSelectedFiles()[0]);
+        } else {
+            QString info = QString::number(sf) + tr(" selected");
+            bar->setMessage(tr("%1 | %2").arg(info, currentDir()));
+            bar->setMessage("", DkStatusBar::status_filesize_info);
+        }
+    } else { // i.e. user STARTED hovering over a file
+        displayFileInfoInStatusbar(filePath);
+    }
+}
 
-        if (sf > 1)
-            info = QString::number(sf) + tr(" selected");
-        else
-            info = QString::number(mThumbLabels.size()) + tr(" images");
+void DkThumbScene::displayFileInfoInStatusbar(const QString &filePath)
+{
+    QFileInfo fInfo(filePath);
+    QString info = QFileInfo(filePath).fileName();
+    DkStatusBar *bar = DkStatusBarManager::instance().statusbar(); 
+    if (fInfo.exists()) {
+        bar->setMessage(DkUtils::readableByte((float)fInfo.size()), DkStatusBar::status_filesize_info);
```

Comment:
```suggestion
        bar->setMessage(DkUtils::readableByte(fInfo.size()), DkStatusBar::status_filesize_info);
```

Int can be converted to float, no need to cast.

Reply:
I changed it here, but the reason I did it like this originally was that I copied the code from `DkNoMacs::setWindowTitle` (currently line 1861 in my branch). Should it not be changed there too then?

Reply:
Sure! You can do it in a separate commit. Thanks!

Reply:
done, as suggested, in a separate commit